src/algorithms/tracking/TracksToParticles.cc

```diff
@@ -46,151 +35,44 @@ namespace eicrecon {
-                rec_assoc.setRecID(rec_part.getObjectID().index);
-                rec_assoc.setSimID((*mc_particles)[best_match].getObjectID().index);
-                rec_assoc.setWeight(1);
-                rec_assoc.setRec(rec_part);
-                auto sim = (*mc_particles)[best_match];
-                rec_assoc.setSim(sim);
-
-                if (level() <= algorithms::LogLevel::kDebug) {
-
-                    const auto &mcpart = (*mc_particles)[best_match];
-                    const auto &p = mcpart.getMomentum();
-                    const auto p_mag = edm4hep::utils::magnitude(p);
-                    const auto p_phi = edm4hep::utils::angleAzimuthal(p);
-                    const auto p_theta = edm4hep::utils::anglePolar(p);
-                    debug(" MCPart: {:<4} {:<8.3f} {:<8.3f} {:<8.2f} {:<6}",
-                                 mcpart.getObjectID().index, p_mag, p_theta, p_phi, mcpart.getCharge(),
-                                 mcpart.getPDG());
-
-                    debug(" Assoc: id={} SimId={} RecId={}",
-                                 rec_assoc.getObjectID().index, rec_assoc.getSim().getObjectID().index, rec_assoc.getSim().getObjectID().index);
-
-                    trace(" Assoc PDGs: sim.PDG | rec.PDG | rec.particleIDUsed.PDG = {:^6} | {:^6} | {:^6}",
-                                 rec_assoc.getSim().getPDG(),
-                                 rec_assoc.getRec().getPDG(),
-                                 rec_assoc.getRec().getParticleIDUsed().isAvailable() ? rec_assoc.getRec().getParticleIDUsed().getPDG() : 0);
-
-
+            double max_weight = -1.;
+            for (auto track_assoc : *track_assocs) {
+                if (track_assoc.getRec() == track) {
+                    trace("Found track association: index={} -> index={}, weight={}",
+                          track_assoc.getRec().getObjectID().index,
+                          track_assoc.getSim().getObjectID().index,
+                          track_assoc.getWeight());
+                    auto part_assoc = part_assocs->create();
+                    part_assoc.setRec(rec_part);
+                    part_assoc.setSim(track_assoc.getSim());
+                    part_assoc.setRecID(part_assoc.getRec().getObjectID().index);
+                    part_assoc.setSimID(part_assoc.getSim().getObjectID().index);
+                    part_assoc.setWeight(track_assoc.getWeight());
+
+                    if (max_weight < track_assoc.getWeight()) {
+                        max_weight = track_assoc.getWeight();
+                        edm4hep::Vector3f referencePoint = {
+                            static_cast<float>(track_assoc.getSim().getVertex().x),
+                            static_cast<float>(track_assoc.getSim().getVertex().y),
+                            static_cast<float>(track_assoc.getSim().getVertex().z)}; // @TODO: not sure if vertex/reference point makes sense here
+                        rec_part.setReferencePoint(referencePoint);
```

Comment:
What's the problem with track.position? https://github.com/eic/EDM4eic/blob/main/edm4eic.yaml#L427

Reply:
Nothing is wrong. I assume, this position will come from vertex in the near future.